model/src/main/scala/explore/data/KeyedIndexedList.scala

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.data
+
+import scala.collection.immutable.ListMap
+
+import cats.implicits._
+import cats.kernel.Eq
+import monocle.Lens
+
+// Each element has a unique Key.
+// Elements can be accessed by Key in O(1).
+// Position in the list (index) can be accessed by Key in O(1).
+case class KeyedIndexedList[K, A] private (private val list: ListMap[K, (A, Int)]) {
```

Comment:
Does `ListMap` give you `O(1)` access time?

Reply:
Apparently not, it stores just a `List` internally. I thought it used a mixed structure, which is what `VectorMap` and `TreeSeqMap` do. I'll switch to the latter, which seems to be more performant in the general case.